### PR TITLE
Fix - Crash due to too big custom Time Signature

### PR DESCRIPTION
**Pull request by Abhay-Varnekar**
Resolves: #12970 

### Summary of Changes:

- **Numerator Field Limit**: Added a maximum value of `999` to the numerator field (`zNominal`) to restrict user input.
- **Graphics Rendering Optimization**: Modified the graphics rendering behavior to avoid continuous re-renders while typing. The graphics now only re-render when the user presses Enter or moves focus away from the numerator field.

- [x] I signed the [CLA](https://musescore.org/en/cla)
- [x] The title of the PR describes the problem it addresses
- [x] Each commit's message describes its purpose and effects, and references the issue it resolves
- [x] If changes are extensive, there is a sequence of easily reviewable commits
- [x] The code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [x] There are no unnecessary changes
- [x] The code compiles and runs on my machine, preferably after each commit individually
- [ ] I created a unit test or vtest to verify the changes I made (if applicable)

### CODE REVIEW

File: src/palette/view/widgets/timedialog.cpp [hunk 1]
```diff
@@ -148,10 +148,14 @@ void TimeDialog::save()
 //   zChanged
 //---------------------------------------------------------
 
-void TimeDialog::zChanged(int val)
+void TimeDialog::zChanged() 
 {
-    Q_UNUSED(val);
-    Fraction sig(zNominal->value(), denominator());
+    int numerator = zNominal->value();  
+    int denominatorValue = this->denominator();
```
Comment (cbjeukendrup) [inline src/palette/view/widgets/timedialog.cpp:154]:
```suggestion
    int denominator = this->denominator();  
```

File: src/palette/view/widgets/timedialog.cpp [hunk 2]
```diff
@@ -148,10 +148,14 @@ void TimeDialog::save()
...
+void TimeDialog::zChanged() 
 {
-    Q_UNUSED(val);
-    Fraction sig(zNominal->value(), denominator());
+    int numerator = zNominal->value();  
+    int denominatorValue = this->denominator();  
+
+    Fraction sig(numerator, denominatorValue);
+
+    // Now update the graphics or palette
```
Comment (cbjeukendrup) [inline src/palette/view/widgets/timedialog.cpp:158]:
```suggestion
    // Update beam groups view
```

File: src/palette/view/widgets/timedialog.cpp [hunk 3]
```diff
@@ -54,7 +54,7 @@ TimeDialog::TimeDialog(QWidget* parent)
     sp->setReadOnly(false);
     sp->setSelectable(true);
 
-    connect(zNominal, &QSpinBox::valueChanged, this, &TimeDialog::zChanged);
+    connect(zNominal,  &QSpinBox::editingFinished, this, &TimeDialog::zChanged);
```
Comment (cbjeukendrup) [inline src/palette/view/widgets/timedialog.cpp:57]: Unnecessary whitespace change:
```suggestion
    connect(zNominal, &QSpinBox::editingFinished, this, &TimeDialog::zChanged);
```

File: src/palette/view/widgets/timedialog.cpp [hunk 4]
```diff
@@ -148,10 +148,14 @@ void TimeDialog::save()
 //   zChanged
 //---------------------------------------------------------
 
-void TimeDialog::zChanged(int val)
+void TimeDialog::zChanged()
```
Comment (cbjeukendrup) [inline src/palette/view/widgets/timedialog.cpp:151]:
```suggestion
void TimeDialog::zChanged()
```

File: src/palette/view/widgets/timedialog.h [hunk 5]
```diff
@@ -70,4 +70,4 @@ private slots:
 };
 }
 
-#endif
+#endif
```
Comment (cbjeukendrup) [inline src/palette/view/widgets/timedialog.h:73]: Another unnecessary whitespace change: you removed the newline at the end of the file

**Review by cbjeukendrup — COMMENTED** [review]
That looks good! A few cleanup suggestions. Also, after implementing these suggestions, it would be great if you could squash the commits so that all changes are in one commit. Thanks!

File: src/palette/view/widgets/timedialog.h (hunk 5 above)
Comment (Abhay-Varnekar) [reply src/palette/view/widgets/timedialog.h]: Extremely sorry as this was my first time working over here messed up a little bit. I will quickly sort it out and create a similar PR
